presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/util/RetryDriver.java

```diff
@@ -81,37 +100,84 @@ private RetryDriver(ListeningScheduledExecutorService executor)
                 Optional.empty(),
                 e -> false,
                 Function.identity(),
-                executor);
+                executor,
+                methodStats,
+                overallStats);
     }
 
-    public static RetryDriver retry(ListeningScheduledExecutorService executor)
+    public static RetryDriver retry(ListeningScheduledExecutorService executor, MBeanExporter mBeanExporter)
     {
-        return new RetryDriver(executor);
+        LoadingCache<String, RetryStats> methodStats = CacheBuilder.newBuilder()
+                .expireAfterAccess(1, TimeUnit.DAYS)
+                .build(new CacheLoader<String, RetryStats>()
+                {
+                    @Override
+                    public RetryStats load(String methodName)
+                            throws Exception
+                    {
+                        RetryStats methodStats = new RetryStats();
+                        String generatedMBeanName = generatedNameOf(RetryStats.class, methodName);
+                        try {
+                            mBeanExporter.unexport(generatedMBeanName);
+                        }
+                        catch (JmxException ignored) {
+                        }
+                        mBeanExporter.export(generatedMBeanName, methodStats);
```

Comment:
Exporting on every call is not the right approach. You want to have a single stats class covering every method which is exported as normal during the Guice binding. Take a look at `ThriftHiveMetastoreStats` and how it is used by `ThriftHiveMetastore`.

Reply:
Thanks for reviewing @electrum , I updated the commit to use a stand alone object for exporting the stats, please let me know if that works better.